osu.Game/Rulesets/Objects/Legacy/LegacyRulesetExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Rulesets.Objects.Types;
+
+namespace osu.Game.Rulesets.Objects.Legacy
+{
+    public static class LegacyRulesetExtensions
+    {
+        /// <summary>
+        /// Introduces floating-point errors to post-multiplied beat length for legacy rulesets that depend on it.
+        /// You should definitely not use this unless you know exactly what you're doing.
+        /// </summary>
+        public static double GetPrecisionAdjustedBeatLength(this IHasSliderVelocity hasSliderVelocity, TimingControlPoint timingControlPoint, string rulesetShortName)
```

Comment:
I still don't like this being an extension method. We've gone from:

`I don't see a reason why rulesets wouldn't use this method, and it'll cause precision loss`
To:
`I don't see a reason why rulesets wouldn't use this method, but it'll crash if they do`

My suggestion was very explicit: have this return `BeatLength * SliderVelocity` for non-legacy rulesets. A second option I would be okay with is this being an explicit static class call `LegacyRulesetExtensions.Get...()`.

I don't know. I don't think I'm being as unreasonable here as I'm being brought to believe. I defer to a 3rd opinion.

Reply:
> I would be okay with is this being an explicit static class call `LegacyRulesetExtensions.Get...()`.

This sounds like a reasonable middle ground... @peppy?

Reply:
Whatever works